Typedoc generation fails for pulumi/pulumi repo

A few findings:
1. Only `typedoc` version 0.8 seems to work for the `pulumi-aws` and `pulumi-cloud`. Any later version (0.9, 0.10, 0.11) will fail with hundreds of errors. I've added the `typedoc` install step in https://github.com/pulumi/docs/pull/100
2. `pulumi/pulumi` fails to typedoc generate with the error below, on all versions of `typedoc` past 0.8. There is a file `version.ts`, so I'm not sure what it's failing on.

```
running typedoc on pulumi
~/go/src/github.com/pulumi/docs ~/go/src/github.com/pulumi/docs
fs.js:1000
  binding.stat(pathModule._makeLong(path), statValues);
          ^

Error: ENOENT: no such file or directory, stat '/Users/donnam/go/src/github.com/pulumi/pulumi/sdk/nodejs/version'
    at Error (native)
    at Object.fs.statSync (fs.js:1000:11)
    at /usr/local/lib/node_modules/typedoc/dist/lib/application.js:140:20
    at Array.forEach (native)
    at CliApplication.Application.expandInputFiles (/usr/local/lib/node_modules/typedoc/dist/lib/application.js:138:20)
    at CliApplication.bootstrap (/usr/local/lib/node_modules/typedoc/dist/lib/cli.js:58:28)
    at CliApplication.Application (/usr/local/lib/node_modules/typedoc/dist/lib/application.js:39:15)
    at new CliApplication (/usr/local/lib/node_modules/typedoc/dist/lib/cli.js:35:42)
    at Object.<anonymous> (/usr/local/lib/node_modules/typedoc/bin/typedoc:4:1)
    at Module._compile (module.js:570:32)
make: *** [generate] Error 1
```

--- a/README.md
+++ b/README.md
@@ -13,7 +13,7 @@ The website is statically built using [Jekyll](https://jekyllrb.com). So we have
 for generating HTML and the ability to write most files in Markdown.
 
 TypeScript documentation is generated directly from source using [TYPEDOC](http://typedoc.org/). We
-just check the resulting files directly into the repo under `./libraries/`.
+just check the resulting files directly into the repo under `./packages/`.
 
 ## Development
 
@@ -22,9 +22,17 @@ installation on your system.
 
 `make build` will generate the website (published to _site).
 
-`make generate` will regenerate the TypeScript documentation if needed. Requires you have the
-other Pulumi repos (pulumi, pulumi-cloud, pulumi-aws) as peers of the docs repo and in a good
-working state.
+`make generate` will regenerate the TypeScript documentation if needed. The generated documentation is placed in the [packages](packages/) folder. This is extremely hacky.
+
+   Prerequisites:
+  - Ensure that repos `pulumi`, `pulumi-aws`, and `pulumi-cloud` are peers to `docs`, and have been checked out to the right release branch.
+  - Run `make only_build` in each of these repos
+  - Install typedoc globally:
+    ```npm install -g typedoc@0.8```
+  - Patch typedoc so it doesn't bring its own version of TypeScript. For more information, see [this issue comment in typedoc](https://github.com/TypeStrong/typedoc/issues/624#issuecomment-352897218). On a Mac:
+    ```rm -rf /usr/local/lib/node_modules/typedoc/node_modules/typescript/```
+  - Install TypeScript globally:
+    ```npm install -g typescript@2.6.2```
 
 `make serve` will build the website and serve it to http://localhost:4000.
 
@@ -34,7 +42,7 @@ working state.
 
 - The mapping from documentation page to section and table-of-contents (TOC) is stored in yaml files in `_data`. 
 
-- To rename a section, rename both the value of `nav_section` in the toc yaml file (e.g., `_data\install.yaml`) and update the layout in `_includes\header.html`.
+- To rename a section, rename both the value of `nav_section` in the toc yaml file (e.g., [`_data/install.yaml`](_data/install.yaml)) and update the layout in [`_includes/header.html`](_includes/header.html).
 
 - To add a new article and add it to the table-of-contents for that section, create a new file in the right section and add an entry in the corresponding file in `_data\top-level-section.yaml`.
 

